internal/csot/csot.go

```diff
@@ -11,26 +11,111 @@ import (
 	"time"
 )
 
-type timeoutKey struct{}
+type withoutMaxTime struct{}
 
-// MakeTimeoutContext returns a new context with Client-Side Operation Timeout (CSOT) feature-gated behavior
-// and a Timeout set to the passed in Duration. Setting a Timeout on a single operation is not supported in
-// public API.
-//
-// TODO(GODRIVER-2348) We may be able to remove this function once CSOT feature-gated behavior becomes the
-// TODO default behavior.
-func MakeTimeoutContext(ctx context.Context, to time.Duration) (context.Context, context.CancelFunc) {
-	// Only use the passed in Duration as a timeout on the Context if it
-	// is non-zero.
-	cancelFunc := func() {}
-	if to != 0 {
-		ctx, cancelFunc = context.WithTimeout(ctx, to)
+// WithoutMaxTime returns a new context with a "withoutMaxTime" value that
+// is used to inform operation construction to not include "maxTimeMS" in a wire
+// message, regardless of a context deadline. This is specifically used for
+// non-awaitable hello commands.
+func WithoutMaxTime(ctx context.Context) context.Context {
+	return context.WithValue(ctx, withoutMaxTime{}, true)
+}
+
+// IsWithoutMaxTime checks if the provided context has been assigned the
+// "withoutMaxTime" value.
+func IsWithoutMaxTime(ctx context.Context) bool {
+	return ctx.Value(withoutMaxTime{}) != nil
+}
+
+type clientLevel struct{}
+
+func AsClientLevel(parent context.Context) context.Context {
+	return context.WithValue(parent, clientLevel{}, true)
+}
+
+func IsClientLevel(ctx context.Context) bool {
```

Comment:
This is only used in the `csot` package, so doesn't need to be exported. It can probably be in-lined in `IsTimeoutContext` since it's only used there.